onnxruntime/core/graph/graph_utils.cc

```diff
@@ -399,9 +399,24 @@ bool IsGraphInput(const Graph& graph, const NodeArg* input) {
   return std::find(graph_inputs.begin(), graph_inputs.end(), input) != graph_inputs.end();
 }
 
+bool IsConstantInitializer(const Graph& graph, const std::string& initializer_name) {
+  bool constant = true;
+
+  if (graph.CanOverrideInitializer()) {
+    const auto& graph_inputs = graph.GetInputsIncludingInitializers();
```

Comment:
>GetInputsIncludingInitializers [](start = 37, length = 30)

Might be cleaner if the function name is GetInputsIncludingOverridableInitializers(). Or maybe GetInputs would do if all overridable initializers are already in graph inputs?

Reply:
What can override an initializer if there's no matching graph input for it? 

---
In reply to: [299592491](https://github.com/microsoft/onnxruntime/pull/1324#discussion_r299592491) [](ancestors = 299592491)

Reply:
I mean for this function's point of view, a name that's neither input nor initializer would be returned as true. It is better to prevent that by having some error checks.

---
In reply to: [299711239](https://github.com/microsoft/onnxruntime/pull/1324#discussion_r299711239) [](ancestors = 299711239,299592491)

Reply:
Glad to see the recursive search. Do we need a testcase with subgraph for it?

---
In reply to: [299744880](https://github.com/microsoft/onnxruntime/pull/1324#discussion_r299744880) [](ancestors = 299744880,299711239,299592491)